crypto/keys/multisig/amino.go

```diff
@@ -78,11 +78,28 @@ func (m *LegacyAminoPubKey) UnmarshalAminoJSON(tmPk tmMultisig) error {
 	// Instead of just doing `*m = *protoPk`, we prefer to modify in-place the
 	// existing Anys inside `m` (instead of allocating new Anys), as so not to
 	// break the `.compat` fields in the existing Anys.
-	m.PubKeys = make([]*types.Any, len(protoPk.PubKeys))
+	if m.PubKeys == nil {
+		m.PubKeys = make([]*types.Any, len(tmPk.PubKeys))
+	}
 	for i := range m.PubKeys {
-		m.PubKeys[i] = &types.Any{}
-		m.PubKeys[i].TypeUrl = protoPk.PubKeys[i].TypeUrl
-		m.PubKeys[i].Value = protoPk.PubKeys[i].Value
+		if m.PubKeys[i] == nil {
```

Comment:
Is `len(m.PubKeys)` the same as `len(tmPk.PubKeys)`? This seems like this case will always evaluate since you've initialized them to nil.

Reply:
Looking at the comment and code intent https://github.com/cosmos/cosmos-sdk/blob/2d90f2be4a94281c7033e8e4400289d6ce248d3f/crypto/keys/multisig/amino.go#L78 `m.Pubkeys` could already be initialized and conceivably have a different length. 
Should I add a `len` check?
```go 
if m.PubKeys[i] == nil && len(m.PubKeys) != len(tmPk.PubKeys){
```
Or why not always fully recreate `m.PubKeys` with the `compat` field? 
The coverage report shows it's always nil entering the function, and if not and there are values mismatches in the Pubkeys structs, it's a problem.
What do you think?

Reply:
I'm not really sure. I find this code to already be pretty confusing tbh. @amaurym do you have any suggestions here?